ocp_resources/machineset.py

```diff
@@ -0,0 +1,91 @@
+from ocp_resources.machine import NamespacedResource
+
+
+class MachineSet(NamespacedResource):
+
+    api_group = NamespacedResource.ApiGroup.MACHINE_OPENSHIFT_IO
+
+    def __init__(
+        self,
+        cluster_name,
+        name=None,
+        namespace=None,
+        teardown=True,
+        client=None,
+        machine_role="worker",
+        machine_type="worker",
+        replicas=1,
+        provider_spec=None,
+        yaml_file=None,
+    ):
+        super().__init__(
+            name=name,
+            namespace=namespace,
+            client=client,
+            teardown=teardown,
+            yaml_file=yaml_file,
+        )
+        self.replicas = replicas
+        self.cluster_name = cluster_name
+        self.machine_role = machine_role
+        self.machine_type = machine_type
+        self.provider_spec = provider_spec if provider_spec else {}
+
+    @property
+    def available_replicas(self):
+        return self.instance.status.availableReplicas
+
+    @property
+    def ready_replicas(self):
+        return self.instance.status.readyReplicas
+
+    @property
+    def desired_replicas(self):
+        return self.instance.status.replicas
+
+    @property
+    def resource_fields(self):
+        return self.instance.spec.template.spec.providerSpec.value
+
+    def to_dict(self):
+
+        data = super().to_dict()
+        if self.yaml_file:
+            return data
+
+        _spec, _selector, _metadata, _labels, _template = (
+            "spec",
+            "selector",
+            "metadata",
+            "labels",
+            "template",
+        )
+        _provider_spec = "providerSpec"
```

Comment:
Why this one is separated from the above?

Reply:
apparently the pre-commit changed that before my commit, ill change that.